src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -164,13 +179,63 @@ impl<'a> EKS<'a> {
+                scope: EngineErrorScope::Engine,
+                execution_id: self.context.execution_id().to_string(),
+                message: Some(format!(
+                    "the number of subnets for zone '{}' should be an even number, not an odd!",
+                    zone_name
+                )),
+            })
+        } else {
+            Ok((subnet_block.len() / 2) as usize)
+        }
+    }
+
     fn tera_context(&self) -> Result<TeraContext, EngineError> {
+        let mut context = TeraContext::new();
+
         let format_ips =
             |ips: &Vec<String>| -> Vec<String> { ips.iter().map(|ip| format!("\"{}\"", ip)).collect::<Vec<_>>() };
 
-        let eks_zone_a_subnet_blocks = format_ips(&self.options.eks_zone_a_subnet_blocks);
-        let eks_zone_b_subnet_blocks = format_ips(&self.options.eks_zone_b_subnet_blocks);
-        let eks_zone_c_subnet_blocks = format_ips(&self.options.eks_zone_c_subnet_blocks);
+        let mut eks_zone_a_subnet_blocks_private = format_ips(&self.options.eks_zone_a_subnet_blocks);
+        let mut eks_zone_b_subnet_blocks_private = format_ips(&self.options.eks_zone_b_subnet_blocks);
+        let mut eks_zone_c_subnet_blocks_private = format_ips(&self.options.eks_zone_c_subnet_blocks);
+
+        if self.options.vpc_qovery_network_mode.is_some() {
+            match self.options.vpc_qovery_network_mode.clone().unwrap() {
+                VpcQoveryNetworkMode::WithNatGateways => {
+                    let max_subnet_zone_a = self.check_odd_subnets("a", &eks_zone_a_subnet_blocks_private)?;
+                    let max_subnet_zone_b = self.check_odd_subnets("b", &eks_zone_b_subnet_blocks_private)?;
+                    let max_subnet_zone_c = self.check_odd_subnets("c", &eks_zone_c_subnet_blocks_private)?;
+
+                    let eks_zone_a_subnet_blocks_public: Vec<String> =
+                        eks_zone_a_subnet_blocks_private.drain(max_subnet_zone_a..).collect();
+                    let eks_zone_b_subnet_blocks_public: Vec<String> =
+                        eks_zone_b_subnet_blocks_private.drain(max_subnet_zone_b..).collect();
+                    let eks_zone_c_subnet_blocks_public: Vec<String> =
+                        eks_zone_c_subnet_blocks_private.drain(max_subnet_zone_c..).collect();
+
+                    context.insert("eks_zone_a_subnet_blocks_public", &eks_zone_a_subnet_blocks_public);
+                    context.insert("eks_zone_b_subnet_blocks_public", &eks_zone_b_subnet_blocks_public);
+                    context.insert("eks_zone_c_subnet_blocks_public", &eks_zone_c_subnet_blocks_public);
+                }
+                VpcQoveryNetworkMode::WithoutNatGateways => {}
+            };
+            context.insert(
+                "vpc_qovery_network_mode",
+                &self.options.vpc_qovery_network_mode.clone().unwrap().to_string(),
+            );
+        } else {
+            context.insert("vpc_qovery_network_mode", &"WithoutNatGateways".to_string());
```

Comment:
Use the enum.to_string(), so if you rename it one day, it will refacto automatically for you